modules/library/main/src/main/java/org/geotools/data/AbstractDataAccessFactory.java

```diff
@@ -0,0 +1,189 @@
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.geotools.data.DataAccessFactory.Param;
+import org.geotools.parameter.DefaultParameterDescriptor;
+import org.geotools.parameter.DefaultParameterDescriptorGroup;
+import org.geotools.parameter.FloatParameter;
+import org.geotools.parameter.Parameter;
+import org.opengis.parameter.ParameterDescriptorGroup;
+import org.opengis.parameter.ParameterValue;
+
+/**
+ * This implementation was lifted from AbstractDataStoreFactory.
+ * As there was nothing in the implementation that was Store-specific (i.e. a restriction on simple types)
+ * this was raised to allow the factory to be used for complex features.
+ * 
+ * @author Adam Brown (Curtin University of Technology)
+ *
+ */
+public abstract class AbstractDataAccessFactory implements DataAccessFactory {
+	 
+    /** Default Implementation abuses the naming convention.
+     * <p>
+     * Will return <code>Foo</code> for
+     * <code>org.geotools.data.foo.FooFactory</code>.
+     * </p>
+     * @return return display name based on class name
+     */
+    public String getDisplayName() {
+        String name = this.getClass().getName();
+        
+        name = name.substring( name.lastIndexOf('.') );
+        if( name.endsWith("Factory")){
+            name = name.substring(0, name.length()-7);
+        } else if( name.endsWith("FactorySpi")){
+            name = name.substring(0, name.length()-10);
+        }
+        return name;
+    }
+    
+    /**
+     * Default implementation verifies the Map against the Param information.
+     * <p>
+     * It will ensure that:
+     * <ul>
+     * <li>params is not null
+     * <li>Everything is of the correct type (or upcovertable
+     * to the correct type without Error)
+     * <li>Required Parameters are present
+     * </ul>
+     * </p>
+     * <p>
+     * <p>
+     * Why would you ever want to override this method?
+     * If you want to check that a expected file exists and is a directory.
+     * </p>
+     * Overrride:
+     * <pre><code>
+     * public boolean canProcess( Map params ) {
+     *     if( !super.canProcess( params ) ){
+     *          return false; // was not in agreement with getParametersInfo
+     *     }
+     *     // example check
+     *     File file = (File) DIRECTORY.lookup( params ); // DIRECTORY is a param
+     *     return file.exists() && file.isDirectory();
+     * }
+     * </code></pre>
+     * @param params
+     * @return true if params is in agreement with getParametersInfo, override for additional checks.
+     */
+    public boolean canProcess( Map params ) {
```

Comment:
This functionality is the only good bit and is available here: http://docs.geotools.org/latest/javadocs/org/geotools/data/DataUtilities.html#canProcess(java.util.Map, org.geotools.data.DataAccessFactory.Param[])